torch_em/data/datasets/pannuke.py

```diff
@@ -0,0 +1,149 @@
+}
+
+
+# TODO
+CHECKSUM = None
+
+
+def _download_pannuke_dataset(path, download, folds):
+    os.makedirs(path, exist_ok=True)
+
+    checksum = CHECKSUM
+
+    for tmp_fold in folds:
+        if os.path.exists(os.path.join(path, f"pannuke_{tmp_fold}.h5")):
+            return
+
+        util.download_source(os.path.join(path, f"{tmp_fold}.zip"), URLS[f"{tmp_fold}"], download, checksum)
+
+        print(f"Unzipping the PanNuke dataset in {tmp_fold} directories...")
+        util.unzip(os.path.join(path, f"{tmp_fold}.zip"), os.path.join(path, f"{tmp_fold}"), True)
+
+
+def _convert_to_hdf5(path, folds):
+    for tmp_fold in folds:
+        if os.path.exists(os.path.join(path, f"pannuke_{tmp_fold}.h5")):
+            return
+
+        print(f"Converting the {tmp_fold} into h5 file format...")
+        tmp_name = tmp_fold.split("_")[0] + tmp_fold.split("_")[1]  # name of a particular sub-directory (per fold)
+        with h5py.File(os.path.join(path, f"pannuke_{tmp_fold}.h5"), "w") as f:
+            img_path = glob(os.path.join(path, tmp_fold, "*", "images", tmp_name, "images.npy"))[0]
+            gt_path = glob(os.path.join(path, tmp_fold, "*", "masks", tmp_name, "masks.npy"))[0]
+
+            f.create_dataset("images", data=np.load(img_path).transpose(3, 0, 1, 2))
+            f.create_dataset("masks", data=np.load(gt_path).transpose(3, 0, 1, 2))
+
+    dir_to_rm = glob(os.path.join(path, "*[!.h5]"))
+    for tmp_dir in dir_to_rm:
+        shutil.rmtree(tmp_dir)
+
+
+def get_pannuke_dataset(
+        path,
+        patch_shape,
+        folds=("fold_1", "fold_2", "fold_3"),
+        rois={},
+        download=False,
+        with_channels=True,
+        with_label_channels=True,
+        **kwargs
+):
+    if rois is not None:
+        assert isinstance(rois, dict)
+
+    _download_pannuke_dataset(path, download, folds)
+    _convert_to_hdf5(path, folds)
+
+    data_paths = [os.path.join(path, f"pannuke_{f_idx}.h5") for f_idx in folds]
+    data_rois = [rois.get(f_idx, np.s_[:, :, :]) for f_idx in folds]
+
+    raw_key = "images"
+    label_key = "masks"
+
+    return torch_em.default_segmentation_dataset(
+        data_paths, raw_key, data_paths, label_key, patch_shape, rois=data_rois,
+        with_channels=with_channels, with_label_channels=with_label_channels, **kwargs
+    )
+
+
+def get_pannuke_loader(
+        path,
+        patch_shape,
+        batch_size,
+        folds=("fold_1", "fold_2", "fold_3"),
+        download=False,
+        rois={},
+        **kwargs
+):
+    """TODO
+    """
+    dataset_kwargs, loader_kwargs = util.split_kwargs(torch_em.default_segmentation_dataset, **kwargs)
+
+    ds = get_pannuke_dataset(
+        path=path,
+        patch_shape=patch_shape,
+        folds=folds,
+        rois=rois,
+        download=download,
+        **dataset_kwargs)
+    return torch_em.get_data_loader(ds, batch_size=batch_size, **loader_kwargs)
+
+
+def label_trafo(labels):
```

Comment:
I don't think this gives you a correct instance segmentation. Even if it does, it's too complex, you don't need the `np.where`. I would write it like this:

```python
segmentation = np.zeros(labels.shape[1:])
max_id = 0
for label_channel in labels[:-1]:  # from what I understand we can just ignore the last channel because it encodes background
    this_labels = vigra.analysis.labelImage(label_channel)  # connected components to make sure we have an instance segmentation
    foreground = this_labels > 0
    segmentation[foreground] = this_labels[foreground] + max_id
    max_id = segmentation.max()
```

Reply:
I refactored the snippets a slight bit with some minor updates (the major fix is on `np.where`)